### PR TITLE
Fix: Recycling dashboard links to labels vs packaging

### PR DESCRIPTION
**Pull request by psankhe28**
Part of #456 

### What
I tried to implement the steps as mentioned in the comment section of the issue #456 .

### Getting this error

image

### CODE REVIEW

**Comment by alexfauquette** [comment]
Thanks for the PR, now it makes sense. You added `type` as a required property (nice 👍)

But now, `DashboardCard` is expected to get `type` in its props. You have two solutions:

- Make `type` optional with `type?: ....` and set a default value (I would not recommend this because there is no default value that is more legitimate than another)
- Add a `type` to every item in `LOGOS`

Are you familiar with shortcuts for fast editing in your IDE?
If not, to avoid spending too much time on this, you can do the following replacement :

Replace `logo: "` by `type: "label", logo: "`

**Comment by abughalib** [comment]
Doing something like this would resolve the issue
```jsx
...
interface Logos {
  [key: string]: LogoDefinition;
}

export const LOGOS: Logos = {
  // Nutriscore
  "en:nutriscore-grade-a": {
    tag: "en:nutriscore-grade-a",
    label: "Nutriscore A",
    logo: "https://static.openfoodfacts.org/images/attributes/nutriscore-a.svg",
    predictor: null,
    type: 'label' // Label
  },
...
```

**Comment by abughalib** [comment]
> @alexfauquette Did the changes by adding type property to LOGOS, but still facing the same error. Can you please check it out once.

Please have a look at my previous message which should fix the issue.
https://github.com/openfoodfacts/hunger-games/pull/457#issuecomment-1372005076

File: src/pages/logosValidator/dashboardDefinition.ts [hunk 1]
```diff
@@ -19,682 +20,707 @@ export type LogoDefinition = {
...
    */
   link?: string;
+  type: InsightType;
   /**
    * To specify if the game needs to be restricted to a specific robotoff predictor
    * @default "universal-logo-detector"
    */
   predictor?: string;
 };
-
```
Comment (abughalib) [inline src/pages/logosValidator/dashboardDefinition.ts:28]: interface Logos {
  [key: string]: LogoDefinition;
}

File: src/pages/logosValidator/dashboardDefinition.ts [hunk 2]
```diff
@@ -19,682 +20,707 @@ export type LogoDefinition = {
...
 export const LOGOS = {
```
Comment (abughalib) [inline src/pages/logosValidator/dashboardDefinition.ts:30]:
```suggestion
export const LOGOS: Logos  = {
```

**Review by abughalib — CHANGES_REQUESTED** [review]
Try those suggested changes, and run Prettier Code Formatter if you can or other if specified.

File: src/pages/logosValidator/dashboardDefinition.ts (hunk 2 above)
Comment (psankhe28) [reply src/pages/logosValidator/dashboardDefinition.ts:30]: Sure. Will try

**Comment by psankhe28** [comment]
It worked. Thanks @abughalib

File: src/pages/logosValidator/dashboardDefinition.ts (hunk 1 above)
Comment (alexfauquette) [reply src/pages/logosValidator/dashboardDefinition.ts:28]: The is that latter on the keys are used to let Typesctip detect speeling mistake and add autocompletion on VSCode
```
logos: (keyof typeof LOGOS)[];
```

With your solution key type is a string, so it breaks this check. I added a commit with the following solution:

```ts
type LogoType = { [key in keyof typeof UNTYPED_LOGOS]: LogoDefinition };

export const LOGOS = UNTYPED_LOGOS as LogoType;
```

**Review by alexfauquette — APPROVED** [review]
Sounds good, I merge, and you can do the 3rd steps in a distinct PR

**Pull request merged**